iwyu_tool.py

```diff
@@ -53,16 +54,17 @@
 GENERAL, ADD, REMOVE, LIST = range(4)
 
 
-def clang_formatter(output, style):
+def clang_formatter(output, style, logLevel):
     """ Process iwyu's output into something clang-like. """
     formatted = []
 
     state = (GENERAL, None)
     for line in output.splitlines():
         match = CORRECT_RE.match(line)
         if match:
-            formatted.append('%s:1:1: note: #includes/fwd-decls are correct' %
-                             match.groups(1))
+            if logLevel is LogLevel.NOTE:
+                formatted.append('%s:1:1: note: #includes/fwd-decls are correct' %
```

Comment:
Actually I'm beginning to wonder if this note is useful at all. I don't use these custom diagnostics myself, so I don't have a feeling for how they behave.

@svenpanne You added this back in 2016, do you have a feeling for how valuable the "everything's OK" diagnostics are? I'm considering if we should just remove them.

Reply:
Well, the underlying original diagnostic _"foobar has correct #includes/fwd-decls"_ is not really helpful, either, see e.g. #949. IIRC I just added the line above to keep a 1:1 correspondence. Personnally, I fully agree with Eric S. Raymond's "Rule of Silence": When a program has nothing surprising to say, it should say nothing.

So in a nutshell: Let's nuke the line mentioned above from both the wrapper and the underlying IWYU, at least in the default config. Adding an option to re-add it might make sense, but OTOH I can't really imagine a scenario where one actually needs that output. Usually e.g. C++ compilers don't tell you that they successfully compiled a file, either. 😉

Reply:
> Let's nuke the line mentioned above from both the wrapper and the underlying IWYU

Oh, I didn't see this. I think this should be added to #1563 if it's not already there, the core IWYU summary diagnostics are pretty terrible, but it takes some careful design to get them better, imho. Let's not do that right now.